### DISCUSSION TITLE
Javascript errors not shown in devtool for browser.EvaluateScriptAsync calls

### DISCUSSION THREAD

**Discussion by holger-prause** [cefsharp/CefSharp#3900] [General]
Hello- i noticed the following strange behaviour, javascript errors are not logged into the devtools window.
Environment: Windows 10
CEF Version: 93.1.140 (retrived via nuget package CefSharp.WinForms)
Embedding Type: Winforms

How to reproduce:
1) create simple testing files
test.html

```
<!DOCTYPE html>
<html>
<head>
	<script type="text/javascript" src="testscript.js"> </script>
</head>
<body>
	<button onclick="doRaiseError()">raise js error</button>
</body>
</html>
```

testscript.js
```
function doRaiseError() {
    console.log("entering funct");
    var a = null;
    a.indexOf("exception");
}
```

Start CEF Browser - open the url and click the "raise js error" button - the error will be shown on the devtools console

```
testscript.js:4 Uncaught TypeError: Cannot read properties of null (reading 'indexOf')
    at doRaiseError (testscript.js:4)
    at HTMLButtonElement.onclick ((index):7)
```

Now reload the page(to be sure) and just call the "doRaiseError" function via browser.EvaluateScriptAsync(methodName, methodArguments); If i inspect the JavascriptResponse object i see: 
"Success" Property to be false and also  the "Message" property which will contain the error
```
Uncaught TypeError: Cannot read properties of null (reading 'indexOf')
@ http://localhost/test/testscript.js:4:6
```

**But the error is not shown in the devtools window**
Is this intended behaviour and do i have a chance to correct this behaviour so that any js errors are always shown in the devtools console?

Disclaimer:
- yes i have devtool for the embedded browser open and running (shown via Browser.ShowDevTools())
- Yes i can just log the exception on host console - but then i loose precious development time (i need js errors in one place)
- No i have not raised any bug tickets on this (i propably will if required)

Thank you very much, if any programming code is required - i can supply it as this problem is pretty easy to reproduce.
Greetings,
Holger

**Comment by amaitland** [comment]
> Is this intended behaviour and do i have a chance to correct this behaviour so that any js errors are always shown in the devtools console?

This behavior is outside our control. You can probably hack something together, if the script was unsuccessful then execute JavaScript to call console.error 

https://developer.mozilla.org/en-US/docs/Web/API/console/error

> No i have not raised any bug tickets on this (i propably will if required)

Please wait for a response before creating an issue in future, thanks.

**Reply by holger-prause** [reply]
Thank you for clarification - i actually ended up with a try / catch / log
